### PR TITLE
🎨 Create Reusable Project Item Component #14

### LINKED ISSUE #14: 🎨 Create Reusable Project Item Component
#  Create reusable project item component
---

Quick links: [Summary](#summary) | [Description](#description) | [Technical-Analysis](#Technical-Analysis) | [Acceptance-Criteria](#Acceptance-Criteria)

---

### Summary

As a user, I want a component I can reuse throughout the site to display project items in a carousel

### Description

This component displays a project item as shown in the design.

Project-item

Your assignment is to implement the component to appear as above.

Please take note of all the design features (including mobile and rounded corners)

#### Link to full design:
```
https://xd.adobe.com/view/b98caf27-6549-49e5-8ed8-c5234788df61-ecdf/
```
### Technical Analysis

This component will be created in `libs/elements/cards` (cards is a module)

The component will be called elewa-project-item-card

The component accepts one Angular input of type:

```typeScript
interface ProjectItem {
  title: string
  description: string
  tag: string
  imgSrc: string
}
```
The type/model should be stored in `libs/models/ui/cards` (cards is a module/library)

The name of the file containing the interface is project-item.interface.ts and it's exported by its parent module above.

### Data

Mock projects are defined for the component and placed in data/sections under the file name highlighted-projects.data.ts.
The file exports a Projects[] with the name "__highlightedProjects".

### Acceptance Criteria

- [ ] When rendered the component is pixel-perfect compared to the design above.
- [ ] The component is clean and does not have more than 50 lines
- [ ] The component SCSS uses CSS variables for colour, font, ... Connected to the main style.scss
- [ ] The code is isolated to the `libs/elements/cards` package unless a justified reason is specified.
- [ ] The model/schema is isolated in the `libs/models/ui/cards` package unless a justified reason is specified.
- [ ] The design is responsive for screen, tablet and phone.
- [ ] The image renders fast and has multiple srcsets.

### PR DESCRIPTION
**Pull request by Ochieng23**
# Description

A component I can reuse throughout the site to display project items in a carousel. 

Fixes #14

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshot (optional)
issue14

## How Has This Been Tested?
Rendered data on the browser. 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/elements/cards/src/lib/elewa-project-item-card/elewa-project-item-card.component.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+<p>elewa-project-item-card works!</p>
```
Comment (LemmyMwaura) [inline libs/elements/cards/src/lib/elewa-project-item-card/elewa-project-item-card.component.html:2]: Remove this P tag. Not needed

File: libs/elements/cards/src/lib/elements-cards.module.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { ElewaProjectItemCardComponent } from './elewa-project-item-card/elewa-project-item-card.component';
+
+@NgModule({
+  imports: [CommonModule],
+  declarations: [ElewaProjectItemCardComponent],
```
Comment (LemmyMwaura) [inline libs/elements/cards/src/lib/elements-cards.module.ts:7]: There are two modules here. The buttons module is the only module that was needed. Cards should just be a folder. So path should be `cards/layout/buttons` with buttons as the module

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]

**Review by LemmyMwaura — COMMENTED** [review]
Kindly share a screenshot of your design.

File: libs/elements/cards/src/lib/elements-cards.module.ts (hunk 2 above)
Comment (Ochieng23) [reply libs/elements/cards/src/lib/elements-cards.module.ts:7]: Technical Analysis
This component will be created in libs/elements/cards (cards is a module) on the instructions. Am now confused.

**Comment by LemmyMwaura** [comment]
This mixes more than one issue. Kindly work on issues separately and then open a PR for each.

**Pull request closed without merging**